# Sikiric Merjem - Graduation Project - Question 12

In [3]:
import scipy.stats as stats
import pandas as pd
import numpy as np

dataset12 = pd.read_csv("Survey__results.csv")
dataset12.head() 

ID Please select your age: Please select your gender:  \
0   1                      24                     Female   
1   2                      19                     Female   
2   3                      19                       Male   
3   4                      18                       Male   
4   5                      20                       Male   

  What is your nationality? What is your current academic year or level?  \
0                   Bosnian                 Senior (fourth year student)   
1                   Bosnian                Freshman (first year student)   
2                   Bosnian                Freshman (first year student)   
3                   Turkish                Freshman (first year student)   
4                   Bosnian                Freshman (first year student)   

  What influenced your choice of field of study in Computer Science?   \
0               Personal interest;Job opportunities;                    
1                                 Personal interest;                    
2                                 Job opportunities;                    
3                                 Personal interest;                    
4                                 Personal interest;                    

       What are you planning to do after graduation?  \
0  Find a job in the industry;Start my own business;   
1           Continue my graduate studies (master’s);   
2                        Find a job in the industry;   
3            Start my own business;I don't know yet;   
4                                  I don't know yet;   

  Please describe “academic career” in your own words. What do you know about academic career? What comes to your mind?  \
0  An academic career is like a journey where peo...                                                                      
1                                  studying for life                                                                      
2                                                  .                                                                      
3     Learning is way more important than a diploma                                                                       
4                                               Good                                                                      

  What do you think are the motivational factors to pursue for an academic career in Computer Science?   \
0  Interest and confidence in research (academic ...                                                      
1  Interest and confidence in research (academic ...                                                      
2                         Flexible work environment;                                                      
3  Job opportunities in universities;Flexible wor...                                                      
4                         Flexible work environment;                                                      

  What do you think are the demotivational factors to pursue academic career in Computer Science?   \
0  Lack of confidence;Lack of information and gui...                                                 
1     Stressful work environment;Lack of confidence;                                                 
2                                                 .;                                                 
3                                Financial concerns;                                                 
4                                Financial concerns;                                                 

  Do you think there is a gender gap in the IT sector in BiH?   \
0                                                Yes             
1                                           Not sure             
2                                           Not sure             
3                                                Yes             
4                                                Yes             

  What are your t

12 - Are you personally concerned about the presence of gender bias in your potential future work environment?

Here we check the relationship between male and female responses on their concerns about the presence of gender bias in their potential future work environment. 
We observe the 'Please select your gender:' and 'Are you personally concerned about the presence of gender bias in your potential future work environment?' fields.

Null and Alternative Hypothesis: 

Ho:For the population of IT students in BiH, gender and the perception of gender bias in their future work environment are not related. 

H1:For the population of IT students in BiH, gender and the perception of gender bias in their future work environment are related.

In [5]:
#number of responses by each gender is summarized in the table
dataset12_table=pd.crosstab(dataset12['Please select your gender:'],dataset12['Are you personally concerned about the presence of gender bias in your potential future work environment?'])
print(dataset12_table)

Are you personally concerned about the presence of gender bias in your potential future work environment?  No  \
Please select your gender:                                                                                      
Female                                                                                                     15   
Male                                                                                                       37   

Are you personally concerned about the presence of gender bias in your potential future work environment?  Not sure  \
Please select your gender:                                                                                            
Female                                                                                                           10   
Male                                                                                                             14   

Are you personally concerned about the presence of gender bias in your

In [6]:
val=stats.chi2_contingency(dataset12_table)
val

Chi2ContingencyResult(statistic=11.616611612593754, pvalue=0.003002512603660512, dof=2, expected_freq=array([[22.48648649, 10.37837838, 15.13513514],
       [29.51351351, 13.62162162, 19.86486486]]))

In [11]:
#Calculate degrees of freedom
nb_rows=len(df.iloc[0:2,0])
nb_cols=len(df.iloc[0,0:3])
dof=(nb_rows-1)*(nb_cols-1)
alpha=0.05

#Observed values = oVal12
#Expected values = eVal12
oVal12=dataset12_table.values
eVal12=val[3]

Chi Square formula: \begin{equation}
x^2 = ∑ (O − E)^2 / E  
\end{equation}

In [19]:
from scipy.stats import chi2
chi_square=sum([(o-e)**2./e for o,e in zip(oVal12, eVal12)])
chi_statistics=chi_square[0]+chi_square[1]+chi_square[2]
print('chi value: ', chi_statistics)

chi value:  11.616611612593754


In [20]:
critical_val=chi2.ppf(q=1-alpha,df=dof)
print('critical value: ', critical_val)

critical value:  5.991464547107979


If Chi-Square value (chi_statistics) is greater than 5.99, reject Ho. (meaning that there is a relationship between gender and perception of gender bias in the IT students' future work environment)

In [21]:
if chi_statistics>=critical_val:
    print("Reject Ho: There is a relationship between gender and perception of gender bias in the IT students' future work environment.")
else:
    print("Accept Ho: There is no relationship between the 2 variables.")

Reject Ho: There is a relationship between gender and perception of gender bias in the IT students' future work environment.


In [25]:
#calculate p-value -alternatively
pVal=1-chi2.cdf(x=chi_statistics,df=dof)
print('p value= ', pVal)
if pVal<=alpha:
    print("Reject Ho: There is a relationship between gender and perception of gender bias in the IT students' future work environment.")
else:
    print("Accept Ho: There is no relationship between the 2 variables.")

p value=  0.0030025126036604677
Reject Ho: There is a relationship between gender and perception of gender bias in the IT students' future work environment.


# Effect Size

In [27]:
    data = np.array([[1, .1, .3, .5],
       [2, .07, .21, .35],
       [3, .06, .17, .29],
       [4, .05,.15,.25],
       [5, .04, .13, .22]])
    sizes = pd.DataFrame(data, columns=['Degrees of Freedom', 'Small Effect', 'Medium Effect', 'Large Effect']) 
    
   # getting the chi sq. stat
    chi2 = val[0]
    # calculating the total number of observations
    n = dataset12_table.sum().sum()
    # getting the degrees of freedom
    dof = min(dataset12_table.shape)-1
    # calculating cramer's v
    v = np.sqrt(chi/(n*dof))
    # printing results
    print(f'V = {v}')
    print(f'Cramer\'s V Degrees of Freedom = {dof}')
    print(f'\nEffect Size Thresholds\n{sizes}\n') 

V = 0.3235029500205768
Cramer's V Degrees of Freedom = 1

Effect Size Thresholds
   Degrees of Freedom  Small Effect  Medium Effect  Large Effect
0                 1.0          0.10           0.30          0.50
1                 2.0          0.07           0.21          0.35
2                 3.0          0.06           0.17          0.29
3                 4.0          0.05           0.15          0.25
4                 5.0          0.04           0.13          0.22



The Cramer’s V result indicates that gender has a medium effect on the perception of gender bias in the IT students' future work environment.